### PR TITLE
[STRAT-91] Heartland Turn Angles and Radii

### PR DESCRIPTION
**Pull request by krishna-kalavadia**
Files in the PR:
- "heartland_turn_angles.csv", contains coordinates of the middle point on each turn and the turn angles 
- "heartland_turn_angles_screenshot.png", shows the vectors from which the angles are calculated 
- "heartland_turn_radii.csv", contains coordinates of the middle point on each turn and the turn radii
- "heartland_turn_radii_screenshot.png", shows the full turning circle for each turn 

The turn angles are also in degrees so let me know if you want me to change that to radians or create another column with radians. 

Hopefully there's no errors in the files  :)

### CODE REVIEW

**Review by e-wai — CHANGES_REQUESTED** [review]
Looks good - can we `heartland_turn_angles.csv` and `heartland_turn_radii.csv` into one CSV (with will have headers ,lat, lon, turn angle (degrees), turn radius (m))?

Also, do you have the script that you used to generate this data?

**Comment by krishna-kalavadia** [comment]
> Looks good - can we combine `heartland_turn_angles.csv` and `heartland_turn_radii.csv` into one CSV (with will have headers ,lat, lon, turn angle (degrees), turn radius (m))?
> 
> Also, do you have the script that you used to generate this data?

Yep, CSV files are merged into one and I'm just fixing up the script to generate this data.

**Comment by krishna-kalavadia** [comment]
Also here is the site that I used to create the turn radii circles:

https://www.mapdevelopers.com/draw-circle-tool.php?circles=%5B%5B16093.4%2C-35%2C-95%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B26.26%2C38.9300809%2C-95.6769466%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B21.85%2C38.9302552%2C-95.6761691%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B40.43%2C38.9305248%2C-95.6749413%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B27.42%2C38.9303725%2C-95.6739033%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B56.69%2C38.9294019%2C-95.6730147%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B196.99%2C38.9269148%2C-95.6739013%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B92.93%2C38.9256744%2C-95.6708139%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B23.24%2C38.9247881%2C-95.6713956%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B73.88%2C38.9208216%2C-95.6715322%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B22.08%2C38.9198761%2C-95.6725851%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B31.37%2C38.9215862%2C-95.6731462%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B62.03%2C38.9210214%2C-95.6739449%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B88.51%2C38.9197936%2C-95.6755235%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B19.29%2C38.9220482%2C-95.6767518%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B38.34%2C38.9226699%2C-95.6761833%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B137.76%2C38.9236689%2C-95.6772924%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%2C%5B182.36%2C38.9263204%2C-95.6744214%2C%22%23AAAAAA%22%2C%22%23000000%22%2C0.4%5D%5D

When you open in up zoom to this address: 38.926320,-95.674421

**Comment by krishna-kalavadia** [comment]
Hey,
I also committed the file that contains the turn coordinates. Each group of three coordinates represents a turn so that from each group of three, two vectors can be created and then you can find the angle between those two vectors

**Review by e-wai — APPROVED** [review]
Some small comments if you have the time. This looks great otherwise, thank you!! 🎆

File: routemodel/routes/Heartland/heartland_turn_angles.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+def gen_vectors(points):
+    """
+    Generates vectors between adjacent coordinates
+    @param points: list of dictionaries of coordinates Ex: [{lat1: long1}, {lat2: long2},... {latN: longN}]
+    @return: list of tuples of vector components Ex: [(x1, y1), (x2, y2) ... (xn, yn)]
+    """
+    vectors = []
+    last_point = (0, 0)
+
+    for index, val in enumerate(points):
```
[the remark this answers is not in the document]
Comment (krishna-kalavadia) [reply routemodel/routes/Heartland/heartland_turn_angles.py:15]: Not sure why I did that, yep removed that!

**Review by e-wai — APPROVED** [review]

**Pull request merged**